### DISCUSSION TITLE
Checkbox List - Relationship option labels

### DISCUSSION THREAD

**Discussion by Fzoltan87** [filamentphp/filament#11426] [Help]
### Package

Form builder

### Package Version

v3.2.16

### How can we help you?

Hi.
How can I make the page names appear recursively between the "Checkbox List" option tags?
I want to display the "Page Model" values inside the "MenuResource" form with the "Checkbox list".

Example:

Database:
```
Schema::create('pages', function (Blueprint $table) {
    $table->id();
    $table->unsignedBigInteger('parent_id')->nullable();
    $table->string('name');
    $table->timestamps();
    $table->foreign('parent_id')->references('id')->on('pages')->onDelete('pages');
});
```

Modell
```
namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Page extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'parent_id'
    ];

    public function parent(): BelongsTo
    {
        return $this->belongsTo(Page::class, 'parent_id');
    }

    public function children(): HasMany
    {
        return $this->hasMany(Page::class, 'parent_id');
    }

    public function descendants()
    {
        return $this->children()->with('descendants');
    }
}
```

MenuResource.php
```
public static function form(Form $form): Form
{
    return $form
        ->schema([
            Forms\Components\TextInput::make('name')->required()->maxLength(255),

            Forms\Components\Toggle::make('is_visible')->required(),

            Forms\Components\CheckboxList::make('items')->label('Pages')->searchable()
                ->options([
                    '1' => 'Home',
                    '2' => 'Car',
                    '3' => 'Car -> Audi',
                    '4' => 'Car -> BMW',
                    '5' => 'Car -> Fiat',
                    '6' => 'Car -> Fiat -> 500',
                ])
        ]);
}
```

**Comment by leandrocfe** [comment]
Sorry, what is the question? You should fill the options using the Page Model to achieve the same result

**Comment by Fzoltan87** [comment]
My question is, how can I make it so that the names of the pages are displayed recursively for the option value of the checkbox list.

I did this manually, but I want to achieve this appearance dynamically.
I just don't know exactly how to query the pages recursively within the options parameter.
![Screenshot 2024-02-15 at 18-57-47 Create Menu - Laravel](https://github.com/filamentphp/filament/assets/16391914/c64b9f31-6897-437d-a8ff-a7cd7766a6ee)

**Reply by micraux** [reply]
I think you can't do that with a checkbox list...

Maybe this can help: [https://filamentphp.com/docs/3.x/forms/advanced#dependant-select-options](https://filamentphp.com/docs/3.x/forms/advanced#dependant-select-options)
